runelite-client/src/main/java/net/runelite/client/util/WorldMapView.java

```diff
@@ -0,0 +1,210 @@
+	private static final int REGION_TILE_SIZE = 64;
+
+	@Getter
+	private WorldPoint center;
+
+	@Getter
+	private float zoom;
+
+	@Getter
+	private int graphicsWidth;
+
+	@Getter
+	private int graphicsHeight;
+
+
+	public void setCenter(WorldPoint center)
+	{
+		this.center = center;
+		updateCalculations();
+	}
+
+	public void setZoom(float zoom)
+	{
+		this.zoom = zoom;
+		updateCalculations();
+	}
+
+	public void setGraphicsWidth(int graphicsWidth)
+	{
+		this.graphicsWidth = graphicsWidth;
+		updateCalculations();
+	}
+
+	public void setGraphicsHeight(int graphicsHeight)
+	{
+		this.graphicsHeight = graphicsHeight;
+		updateCalculations();
+	}
+
+	private int regionSizeScaled;
+	private Point tileOffset;
+	@Getter
+	private float pixelsPerTile;
+	private Point graphicalOffset;
+	private Point graphicalCenter;
+
+	private void updateCalculations()
+	{
+		//Calculate region graphical size.
+		regionSizeScaled = (int) (REGION_GRAPHIC_SIZE * zoom);
+		//How far from top left of region this tile is
+		tileOffset = new Point(center.getX() % REGION_TILE_SIZE, center.getY() % REGION_TILE_SIZE);
+		//Graphical size of tileOffset
+		pixelsPerTile = regionSizeScaled / (float)REGION_TILE_SIZE;
+		graphicalOffset = new Point((int)(tileOffset.x * pixelsPerTile), (int)(tileOffset.y * pixelsPerTile));
+		graphicalCenter = new Point(graphicsWidth / 2, graphicsHeight / 2);
+	}
+
+
+	@Inject
+	public WorldMapView()
+	{
+		if (regions == null)
+		{
+			regions = CacheBuilder.newBuilder()
+				.maximumSize(1685)
+				.expireAfterAccess(1, TimeUnit.MINUTES)
+				.build(
+					new CacheLoader<Integer, BufferedImage>()
+					{
+						public BufferedImage load(Integer key) throws IOException
+						{
+							return ImageIO.read(WorldMapView.class.getResourceAsStream("/world_map/img-0-" + key + ".png"));
+						}
+					});
+		}
+	}
+
+	private static BufferedImage getMapRegion(Point worldPoint)
+	{
+		try
+		{
+			int x = worldPoint.x;
+			int y = worldPoint.y;
+			int regionX = x >> 6;
+			int regionY = y >> 6;
+			int regionId = regionX << 8 | regionY;
+			return regions.get(regionId);
+		}
+		catch (ExecutionException e)
+		{
+			return null;
+		}
+	}
+
+	public Point[] worldPointsToGraphicsPoint(WorldPoint[] worldPoints)
+	{
+		Point[] retPoints = new Point[worldPoints.length];
+		for (int i = 0; i < worldPoints.length; i++)
+		{
+			retPoints[i] = worldPointToGraphicsPoint(worldPoints[i]);
+		}
+		return  retPoints;
+	}
+
+	/**
+	 *
+	 * @param worldPoint A WorldPoint to calculate the graphical position for
+	 * @return the graphical coordinates of the specified point.
+	 */
+	public Point worldPointToGraphicsPoint(WorldPoint worldPoint)
+	{
+		Point difference = new Point((int) ((worldPoint.getX() - center.getX()) * pixelsPerTile), (int) ((center.getY() - worldPoint.getY()) * pixelsPerTile));
+		difference.translate(graphicalCenter.x, graphicalCenter.y);
+		return difference;
+	}
+
+	public BufferedImage getMapImage()
```

Comment:
Probably stop using getter and setter-like naming for methods that are neither. Name this buildMapImage or something like that.